content/docs/command-reference/gc.md

```diff
@@ -1,60 +1,75 @@
 # gc
 
-Remove unused objects from <abbr>cache</abbr> or remote storage.
+Remove unused files and directories from <abbr>cache</abbr> or
+[remote storage](/doc/command-reference/remote).
 
 ## Synopsis
 
 ```usage
-usage: dvc gc [-h] [-q | -v] [-a] [-T] [-c] [-r <name>]
+usage: dvc gc [-h] [-q | -v]
+              [-w] [-a] [-T] [--all-commits] [-c] [-r <name>]
               [-f] [-j <number>] [-p [<path> [<path> ...]]]
 ```
 
 ## Description
 
-This command deletes (garbage collects) data files or directories that may exist
-in the cache (or [remote storage](/doc/command-reference/remote) if `-c` is
-used) but no longer referenced in [DVC-files](/doc/user-guide/dvc-file-format)
-currently in the <abbr>workspace</abbr>. By default, this command only cleans up
-the local cache, which is typically located on the same machine as the project
-in question. This usually helps to free up disk space.
+This command deletes (garbage collects) data files or directories that exist in
+DVC cache but are no longer needed. With `--cloud` it also removes data in
+[remote storage](/doc/command-reference/remote).
 
-There are important things to note when using Git to version the
-<abbr>project</abbr>:
+To avoid accidentally deleting data, it raises an error and doesn't touch any
+files if no scope options are provided. It means it's user's responsibility to
+explicitly provide the right set of options to specify what data is still in-use
+(so that DVC can figure out what fils can be safely deleted).
 
-- If the cache/remote holds several versions of the same data, all except the
-  current one will be deleted.
-- Use the `--all-branches` or `--all-tags` options to avoid collecting data
-  referenced in the tips of all branches or all tags, respectively.
+One of the scope options - `--workspace`, `--all-branches`, `--all-tags`, or
+`--all-commits`, or any combination of them must be provided. Each of them
+corresponds to the current workspace _and_ a set of commits to analyze what
+files, directories and what versions are still needed and should be kept (by
+analyzing DVC-files in those commits).
```

Comment:
```suggestion
`--all-commits`, or any combination of them must be provided. Each of them
corresponds to keeping the data for the current workspace and for a
different set of commits (determined by reading the DVC-files in them).
```

Reply:
Question: what's the point of this option? To me it sounds like "don't gc anything" except maybe deleted parts of the repo like removed branches, rewritten commits, etc.

Reply:
DVC cache contains data that is not referenced in any way in Git history. E.g. run do a lot of `dvc run`, or `dvc repro` after modifying certain files/settings. If `--no-commit` is not specified DVC is saving data to cache not matter you do `git commit` to save DVC-files or not.

Reply:
We probably mention this.

Reply:
mentioned a possible use case in the recent version - @jorgeorpinel pls take a look

Reply:
Makes sense, thanks.

Reply:
Still this suggestion wasn't applied but I'll address in regular updates.

Reply:
I put a new paragraph that explains the use case, have you seen it? Or may be I didn't get you specific suggestion ... 🤔

Reply:
I see it, thanks. May reword a little in regular updates but it's great!

> One of the use cases for this option is to safely delete all temporary data DVC cached when dvc run and/or dvc repro were run without committing changes to DVC-files (thus potentially caching data that is not referenced from workspace or Git commits).